<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MotorControl</class>
 <widget class="QWidget" name="MotorControl">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>440</width>
    <height>290</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>440</width>
    <height>290</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>440</width>
    <height>290</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>MotorControl</string>
  </property>
  <property name="windowIcon">
   <iconset theme="go-home"/>
  </property>
  <property name="styleSheet">
   <string notr="true">

/* 전체 위젯 스타일 */
/* 전체 위젯 스타일 */
QWidget {
    font: 9pt &quot;나눔고딕&quot;;
    background-color: rgb(255, 255, 255);
}

/* 버튼 스타일 */
QPushButton {
    background-color: lightgray; /* 버튼 배경색 */
    border: none; /* 테두리 제거 */
    color: white; /* 텍스트 색상 */
    height: 30px; /* 높이 설정 */
    text-align: center; /* 텍스트 가운데 정렬 */
    font-size: 14px; /* 폰트 크기 */
    transition-duration: 0.4s; /* 전환 효과 지속 시간 */
    cursor: pointer; /* 마우스 커서 포인터로 변경 */
    border-radius: 8px; /* 버튼 모서리 둥글게 */
    box-shadow: 0 4px 8px 0 rgba(0,0,0,0.2); /* 그림자 효과 */
}

QPushButton:hover {
    background-color: #848484; /* 호버 시 배경색 변경 */
}

QPushButton:pressed {
    background-color: #4CAF50; /* 클릭 시 배경색 변경 */
    box-shadow: 0 2px 4px rgba(0,0,0,0.2); /* 그림자 크기 감소 */
}

/* 콤보박스 스타일 */
QComboBox {
    background-color: #ffffff; /* 배경색 */
    border: 1px solid #cccccc; /* 테두리 */
    border-radius: 5px; /* 테두리 모서리 둥글게 */
    padding: 5px; /* 내부 여백 */
    selection-background-color: #e0e0e0; /* 선택된 항목 배경색 */
}

QComboBox QAbstractItemView {
    background-color: #ffffff; /* 배경색 */
    border: 1px solid #cccccc; /* 테두리 */
    selection-background-color: #e0e0e0; /* 선택된 항목 배경색 */
}

QComboBox::down-arrow {
    image: url(down.png);
    width: 10px; /* 이미지 너비 */
    height: 10px; /* 이미지 높이 */
}

QComboBox::drop-down {
    subcontrol-origin: padding; /* 드롭다운 버튼 위치 */
    subcontrol-position: right center; /* 드롭다운 버튼 위치 */
    width: 20px; /* 드롭다운 버튼 너비 */
    border-left-width: 1px; /* 드롭다운 버튼 왼쪽 테두리 너비 */
    border-left-color: #cccccc; /* 드롭다운 버튼 왼쪽 테두리 색상 */
    border-left-style: solid; /* 드롭다운 버튼 왼쪽 테두리 스타일 */
    background-color: #ffffff; /* 드롭다운 버튼 배경색 */
}

</string>
  </property>
  <widget class="QWidget" name="horizontalLayoutWidget">
   <property name="geometry">
    <rect>
     <x>30</x>
     <y>10</y>
     <width>381</width>
     <height>49</height>
    </rect>
   </property>
   <layout class="QHBoxLayout" name="horizontalLayout_5">
    <property name="spacing">
     <number>10</number>
    </property>
    <item>
     <widget class="QLabel" name="label">
      <property name="maximumSize">
       <size>
        <width>30</width>
        <height>16777215</height>
       </size>
      </property>
      <property name="text">
       <string>포트</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
      <property name="buddy">
       <cstring>label</cstring>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QComboBox" name="cb_port">
      <property name="focusPolicy">
       <enum>Qt::WheelFocus</enum>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QLabel" name="label_2">
      <property name="maximumSize">
       <size>
        <width>30</width>
        <height>16777215</height>
       </size>
      </property>
      <property name="text">
       <string>속도</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QComboBox" name="cb_speed">
      <property name="focusPolicy">
       <enum>Qt::WheelFocus</enum>
      </property>
      <property name="contextMenuPolicy">
       <enum>Qt::NoContextMenu</enum>
      </property>
      <property name="editable">
       <bool>false</bool>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QPushButton" name="pb_open">
      <property name="styleSheet">
       <string notr="true"/>
      </property>
      <property name="text">
       <string>열기</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QPushButton" name="pb_close">
      <property name="text">
       <string>닫기</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QFrame" name="frame">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>80</y>
     <width>421</width>
     <height>191</height>
    </rect>
   </property>
   <property name="frameShape">
    <enum>QFrame::StyledPanel</enum>
   </property>
   <property name="frameShadow">
    <enum>QFrame::Sunken</enum>
   </property>
   <widget class="QWidget" name="verticalLayoutWidget_2">
    <property name="geometry">
     <rect>
      <x>160</x>
      <y>10</y>
      <width>251</width>
      <height>171</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_3">
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_11" stretch="0,0,2,0,1">
       <property name="spacing">
        <number>6</number>
       </property>
       <property name="leftMargin">
        <number>5</number>
       </property>
       <item alignment="Qt::AlignLeft">
        <widget class="QLabel" name="label_12">
         <property name="maximumSize">
          <size>
           <width>40</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="text">
          <string>RPM</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="horizontalSpacer">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeType">
          <enum>QSizePolicy::Fixed</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>10</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QComboBox" name="cb_rpm">
         <property name="focusPolicy">
          <enum>Qt::WheelFocus</enum>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="horizontalSpacer_2">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeType">
          <enum>QSizePolicy::Fixed</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>10</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QPushButton" name="pb_set">
         <property name="text">
          <string>설정</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <spacer name="verticalSpacer">
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
       <property name="sizeType">
        <enum>QSizePolicy::Fixed</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>20</width>
         <height>10</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_12">
       <property name="spacing">
        <number>6</number>
       </property>
       <item>
        <widget class="QPushButton" name="pb_up">
         <property name="text">
          <string>UP</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="pb_stop">
         <property name="text">
          <string>STOP</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="pb_down">
         <property name="text">
          <string>DOWN</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="pb_zero">
         <property name="text">
          <string>영점 세팅</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </widget>
   <widget class="QFrame" name="frame_2">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>141</width>
      <height>171</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::Box</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
    <widget class="QWidget" name="verticalLayoutWidget">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>10</y>
       <width>121</width>
       <height>151</height>
      </rect>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_6">
        <item>
         <widget class="QLabel" name="label_5">
          <property name="styleSheet">
           <string notr="true">QLineEdit{border-radius:5}</string>
          </property>
          <property name="text">
           <string>전원 상태 : </string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="lb_power">
          <property name="text">
           <string/>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_7">
        <item>
         <widget class="QLabel" name="label_6">
          <property name="text">
           <string>동작 방향 : </string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="lb_locate">
          <property name="text">
           <string/>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_9">
        <item>
         <widget class="QLabel" name="label_10">
          <property name="text">
           <string>이동 거리 : </string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="lb_move">
          <property name="text">
           <string/>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </widget>
  </widget>
 </widget>
 <tabstops>
  <tabstop>cb_port</tabstop>
  <tabstop>cb_speed</tabstop>
  <tabstop>pb_open</tabstop>
  <tabstop>pb_close</tabstop>
  <tabstop>cb_rpm</tabstop>
  <tabstop>pb_set</tabstop>
  <tabstop>pb_up</tabstop>
  <tabstop>pb_stop</tabstop>
  <tabstop>pb_down</tabstop>
  <tabstop>pb_zero</tabstop>
 </tabstops>
 <resources/>
 <connections/>
</ui>
